### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by riccardocadei**
Update documentation according to the new paper version (e.g. rename Causal Rules with Rules).
Rename functions (and .R files):
- `interpret_select_rules` in `interpret_rules`,
- `generate_causal_rules` in `discover_rules`,
- `discover_causal_rules` in `select_rules`.

### CODE REVIEW

**Comment by riccardocadei** [comment]
@Naeemkh Why are we getting different results on different OS?

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # CRE (Developing)
 
 ## Changed
+* Rename `interpret_select_rules` in `interpret_rules`.
```
Comment (Naeemkh) [inline NEWS.md:4]: `interpret_select_rules` -> `interpret_rules` and also others.

File: R/discover_rules.R [hunk 2]
```diff
@@ -1,23 +1,24 @@
 #' @title
-#' Generate causal rules
+#' Discover Rules
```
Comment (Naeemkh) [inline R/discover_rules.R:2]: Discover **r**ules

File: R/discover_rules.R [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
 #'
 #' @description
-#' Generates causal decision rules.
+#' Discover the minimal set of Rules Linearly Decomposing the Conditional
```
Comment (Naeemkh) [inline R/discover_rules.R:5]: `Rules Linearly Decomposing` please change to small case letters.

File: R/discover_rules.R [hunk 4]
```diff
@@ -1,23 +1,24 @@
...
+#' Average Treatment Effect (CATE).
 #'
 #' @param X A covariate matrix.
 #' @param ite An estimated ITE.
 #' @param method_params A vector of method parameters.
 #' @param hyper_params A vector of hyper parameters.
 #'
 #' @return
-#' A vector of causal rules.
+#' The minimal set of Rules Linearly Decomposing the CATE.
```
Comment (Naeemkh) [inline R/discover_rules.R:14]: A minimal set of rules linearly decomposing the Conditional Treatment Effect (CATE).

File: R/discover_rules.R [hunk 5]
```diff
@@ -32,7 +33,7 @@ generate_causal_rules <- function(X, ite, method_params, hyper_params) {
   # Filtering ------------------------------------------------------------------
 
   # Discard irrelevant variable-value pair from a rule condition ---------------
-  logger::log_info("Filtering irrelevant rules...")
+  logger::log_info("Filtering Irrelevant Rules...")
```
Comment (Naeemkh) [inline R/discover_rules.R:36]: Why `Irrelevant Rules` are in capital?

File: R/discover_rules.R [hunk 6]
```diff
@@ -42,35 +43,35 @@ generate_causal_rules <- function(X, ite, method_params, hyper_params) {
   rules_matrix <- generate_rules_matrix(X, rules_list)
 
   # Discard rules with too few or too many observations and correlated rules ---
-  logger::log_info("Filtering extreme rules...")
+  logger::log_info("Filtering Extreme Rules...")
```
Comment (Naeemkh) [inline R/discover_rules.R:46]: Same here.

File: R/discover_rules.R [hunk 7]
```diff
@@ -42,35 +43,35 @@ generate_causal_rules <- function(X, ite, method_params, hyper_params) {
...
   rules_matrix <- filter_extreme_rules(rules_matrix, rules_list,
                                        getElement(hyper_params, "t_ext"))
   rules_list <- colnames(rules_matrix)
   M_filter2 <- length(rules_list)
 
   # Discard correlated rules ---------------------------------------------------
-  logger::log_info("Filtering correlated rules...")
+  logger::log_info("Filtering Correlated Rules...")
```
Comment (Naeemkh) [inline R/discover_rules.R:53]: Also here.

File: R/filter_correlated_rules.R [hunk 8]
```diff
@@ -1,17 +1,17 @@
 #' @title
-#' Filter correlated decision rules
+#' Filter correlated rules
 #'
 #' @description
-#' Discard higly correlated decision rules (i.e. `Cov(rule1,rule2) > t_corr`).
+#' Discard highly correlated rules (i.e. `Cov(rule1,rule2) > t_corr`).
```
Comment (Naeemkh) [inline R/filter_correlated_rules.R:5]: Discard -> Discards

File: R/generate_rules.R [hunk 9]
```diff
@@ -2,8 +2,9 @@
 #' Generate rules
 #'
 #' @description
-#' Generates a set of relevant decision rules characterizing the
-#' heterogeneity in the ITE.
+#' Generate a list of Rules characterizing the heterogeneity in the Conditional
+#' Average Treatment Effect (CATE) by tree-based metods (Random Forest and
+#' XGBoost).
```
Comment (Naeemkh) [inline R/generate_rules.R:7]: Either: `random forest` or `randomForest`

File: R/select_rules.R [hunk 10]
```diff
@@ -1,29 +1,30 @@
...
-#' falsely selected rules).
-#' @param penalty_rl Order of penalty for rules length during LASSO for Causal
-#' Rules Discovery (i.e. 0: no penalty, 1: rules_length, 2: rules_length^2)
+#' falsely selected rules). Only for Stability Selection.
+#' @param penalty_rl Order of penalty for rules length during LASSO
+#' regularization (i.e. 0: no penalty, 1: rules_length, 2: rules_length^2).
 #'
 #' @return
-#' List of the Causal Decision Rules discovered
+#' The minimal set of Rules Linearly Decomposing the CATE.
```
Comment (Naeemkh) [inline R/select_rules.R:21]: A minimal set of ...
Also unnecessary capitalization.

File: tests/testthat/test-estimate_cate.R [hunk 11]
```diff
@@ -368,7 +368,7 @@ test_that("CATE (linreg) Estimation Runs Correctly", {
     select_rules_interpretable <- NA
   } else {
     rules_matrix_inf <- generate_rules_matrix(X_inf, select_rules_dis)
-    select_rules_interpretable <- interpret_select_rules(select_rules_dis,
+    select_rules_interpretable <- interpret_rules(select_rules_dis,
```
[the remark this answers is not in the document]
Comment (Naeemkh) [reply tests/testthat/test-estimate_cate.R:371]: And rest of them.

**Review by Naeemkh — CHANGES_REQUESTED** [review]
Thanks for the PR. I put a couple of minor comments. Please fix them and resubmit.

**Comment by Naeemkh** [comment]
> @Naeemkh Why are we getting different results on different OS?

Where did that happen?

File: NEWS.md (hunk 1 above)
Comment (riccardocadei) [reply NEWS.md:4]: We never used this convention in the previous updates

File: R/discover_rules.R (hunk 2 above)
Comment (riccardocadei) [reply R/discover_rules.R:2]: addressed in the new PR

File: R/discover_rules.R (hunk 3 above)
Comment (riccardocadei) [reply R/discover_rules.R:5]: addressed in the new PR

File: R/discover_rules.R (hunk 4 above)
Comment (riccardocadei) [reply R/discover_rules.R:14]: addressed in the new PR

**Pull request merged**